src/server/pkg/work/work.go

```diff
@@ -267,7 +268,7 @@ func NewWorker(etcdClient *etcd.Client, etcdPrefix string, taskNamespace string)
 }
 
 // ProcessFunc is a callback that is used for processing a subtask in a task.
-type ProcessFunc func(context.Context, *Task) error
+type ProcessFunc func(context.Context, *Task) (*types.Any, error)
```

Comment:
I think we would be better off having one proto for a particular task, and having result / status information as a part of the proto (another field, a oneof with a task / result, or whatever). Each additional proto type introduced in the work package will require a separate collection that needs to be managed.